testproject/Assets/Tests/Runtime/NetworkSceneManager/NetworkObjectSceneMigrationTests.cs

```diff
@@ -0,0 +1,510 @@
+                serverNetworkObject.Spawn();
+                m_ServerSpawnedPrefabInstances.Add(serverNetworkObject);
+            }
+            yield return WaitForConditionOrTimeOut(VerifyAllClientsSpawnedInstances);
+            AssertOnTimeout($"Timed out waiting for all clients to spawn {nameof(NetworkObject)}s!");
+
+            // Now load three scenes to migrate the newly spawned NetworkObjects into
+            m_ServerNetworkManager.SceneManager.OnSceneEvent += SceneManager_OnSceneEvent;
+            for (int i = 0; i < 3; i++)
+            {
+                m_ClientsLoadedScene = false;
+                m_CurrentSceneLoading = m_TestScenes[i];
+                var status = m_ServerNetworkManager.SceneManager.LoadScene(m_TestScenes[i], LoadSceneMode.Additive);
+                Assert.True(status == SceneEventProgressStatus.Started, $"Failed to start loading scene {m_CurrentSceneLoading}! Return status: {status}");
+                yield return WaitForConditionOrTimeOut(() => m_ClientsLoadedScene);
+                AssertOnTimeout($"Timed out waiting for all clients to load scene {m_CurrentSceneLoading}!");
+            }
+
+            var objectCount = 0;
+            // Migrate each networkObject into one of the three scenes.
+            // There will be 3 networkObjects per newly loaded scenes when done.
+            foreach (var scene in m_ScenesLoaded)
+            {
+                // Now migrate the NetworkObject
+                SceneManager.MoveGameObjectToScene(m_ServerSpawnedPrefabInstances[objectCount].gameObject, scene);
+                SceneManager.MoveGameObjectToScene(m_ServerSpawnedPrefabInstances[objectCount + 1].gameObject, scene);
+                SceneManager.MoveGameObjectToScene(m_ServerSpawnedPrefabInstances[objectCount + 2].gameObject, scene);
+                objectCount += 3;
+            }
+
+            yield return WaitForConditionOrTimeOut(VerifySpawnedObjectsMigrated);
+            AssertOnTimeout($"Timed out waiting for all clients to migrate all NetworkObjects into the appropriate scenes!");
+
+            // Verify that a late joining client synchronizes properly
+            yield return CreateAndStartNewClient();
+            yield return WaitForConditionOrTimeOut(VerifySpawnedObjectsMigrated);
+            AssertOnTimeout($"[Late Joined Client] Timed out waiting for all clients to migrate all NetworkObjects into the appropriate scenes!");
+        }
+
+        /// <summary>
+        /// Integration test to verify changing the currently active scene
+        /// will migrate NetworkObjects with ActiveSceneSynchronization set
+        /// to true.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator ActiveSceneSynchronizationTest()
+        {
+            // Disable resynchronization for this test to avoid issues with trying
+            // to synchronize them.
+            NetworkSceneManager.DisableReSynchronization = true;
```

Comment:
This is integration test specific and is not needed under normal operating conditions (i.e. runtime or in editor).